node/node.go

```diff
@@ -792,34 +795,21 @@ func (n *Node) CreateTransaction(
 }
 
 // GetAccount will return account details for a given address
-func (n *Node) GetAccount(address string) (api.AccountResponse, error) {
-	if check.IfNil(n.coreComponents.AddressPubKeyConverter()) {
-		return api.AccountResponse{}, ErrNilPubkeyConverter
-	}
-	if check.IfNil(n.stateComponents.AccountsAdapterAPI()) {
-		return api.AccountResponse{}, ErrNilAccountsAdapter
-	}
-
-	addr, err := n.coreComponents.AddressPubKeyConverter().Decode(address)
-	if err != nil {
-		return api.AccountResponse{}, err
-	}
-
-	accWrp, err := n.stateComponents.AccountsAdapterAPI().GetExistingAccount(addr)
+func (n *Node) GetAccount(address string, options api.AccountQueryOptions) (api.AccountResponse, api.BlockInfo, error) {
+	account, blockInfo, err := n.loadUserAccountHandlerByAddress(address, options)
 	if err != nil {
 		if err == state.ErrAccNotFound {
 			return api.AccountResponse{
 				Address:         address,
 				Balance:         "0",
 				DeveloperReward: "0",
-			}, nil
+			}, api.BlockInfo{}, nil
+		}
+		if err == ErrCannotCastAccountHandlerToUserAccountHandler {
+			return api.AccountResponse{}, api.BlockInfo{}, errors.New("account is not of type with balance and nonce")
```

Comment:
`"account is not of type with balance and nonce`" is misleading. maybe return the same error that you check for. ALso, add a log error as it would be a programming error

Reply:
Fixed. Added a `log.Error()` as well.